rrmngmnt/nmcli.py

```diff
@@ -0,0 +1,644 @@
+            {"+ipv4.addresses": "192.168.23.2"}, or a '-' in order to remove
+            a property.
+        """
+        command = "nmcli dev modify {device}".format(device=device)
+
+        for k, v in properties.items():
+            command += " {k} {v}".format(k=k, v=v)
+
+        self._exec_command(command=command)
+
+    @staticmethod
+    def _ip_options_builder(
+        ipv4_addr, ipv4_gw, ipv4_method, ipv6_addr, ipv6_gw, ipv6_method
+    ):
+        """
+        Extends a connection adding command with optional parameters.
+
+        Args:
+            ipv4_addr (str): a static address.
+            ipv4_gw (str): a gateway address.
+            ipv4_method (str): setting method.
+                Available methods: auto, disabled, link-local, manual, shared.
+            ipv6_addr (str): a static address.
+            ipv6_gw (str): a gateway address.
+            ipv6_method (str): setting method.
+                Available methods: auto, dhcp, disabled, ignore, link-local,
+                manual, shared.
+
+        Returns:
+            str: an nmcli connection add command with the passed in optional
+                parameters.
+        """
+        command = ""
+
+        if ipv4_method:
+            command += " ipv4.method {method}".format(method=ipv4_method)
+        if ipv6_method:
+            command += " ipv6.method {method}".format(method=ipv6_method)
+        if ipv4_addr:
+            command += " ipv4.addresses {ipv4_addr}".format(ipv4_addr=ipv4_addr)  # noqa: E501
+        if ipv4_gw:
+            command += " ipv4.gateway {ipv4_gw}".format(ipv4_gw=ipv4_gw)
+        if ipv6_addr:
+            command += " ipv6.addresses {ipv6_addr}".format(ipv6_addr=ipv6_addr)  # noqa: E501
+        if ipv6_gw:
+            command += " ipv6.gateway {ipv6_gw}".format(ipv6_gw=ipv6_gw)
+        return command
+
+    @staticmethod
+    def _common_options_builder(
+        con_type, con_name, ifname, auto_connect=None, save=None
+    ):
+        """
+        Generates a string containing common options for the nmcli tool.
+
+        Args:
+            con_type (str): the connection type.
+            con_name (str): the created connection's name.
+            ifname (str): the interface name to use.
+            auto_connect (bool): True to connect automatically, or False for
+                manual.
+            save (bool): True to persist the connection, or False.
+
+        Returns:
+            str: a common options string.
+        """
+        common_options = (
+            "type {type} " "con-name {con_name} " "ifname {ifname} "
+        ).format(type=con_type, con_name=con_name, ifname=ifname)
+        if auto_connect is not None:
+            common_options += "autoconnect {val}".format(
+                val="yes" if auto_connect is True else "no"
+            )
+        if save is not None:
+            common_options += "save {val}".format(val="yes" if save is True else "no")  # noqa: E501
```

Comment:
"save {val}".format(val="yes" if save is True else "no")

extract to function, it is used twice
def _get_str_value(value_):
    return "yes" if value_ is True else "no"